plugins/uefi/fu-plugin-uefi.c

```diff
@@ -413,6 +414,23 @@ fu_plugin_update (FuPlugin *plugin,
 	if (re == NULL)
 		return FALSE;
 
+	/* test the flash counter */
+	flashes_left = fu_device_get_flashes_left (device);
+	if (flashes_left > 0) {
+		g_debug ("%s has %" G_GUINT32_FORMAT " flashes left",
+			 fu_device_get_name (device),
+			 flashes_left);
+		if ((flags & FWUPD_INSTALL_FLAG_FORCE) == 0 && flashes_left <= 2) {
+			g_set_error (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOT_SUPPORTED,
+				     "%s only has %" G_GUINT32_FORMAT " flashes left -- "
+				     "see https://github.com/hughsie/fwupd/wiki/Dell-TPM:-flashes-left for more information.",
```

Comment:
Since this is generic now, we'll probably need to make a new page for this.

Reply:
Agreed; I think we can fix this later, no panic.